library(forcats)
library(ggplot2)
data <- data.frame(y = c(287.5,
                         313.071428571429,
                         254.142857142857,
                         89.9285714285714,
                         64.2857142857143,
                         34.8571428571429,
                         25.6428571428571,
                         15.8571428571429,
                         10,
                         245.5,
                         157.071428571429,
                         249.642857142857,
                         231.571428571429,
                         79.7142857142857,
                         151.428571428571,
                         105.928571428571,
                         13.9285714285714,
                         4.85714285714286,
                         1,
                         79.2142857142857,
                         76.4285714285714,
                         35.2857142857143,
                         35.1428571428571,
                         48,
                         116.071428571429,
                         0,
                         59.3571428571429,
                         17.3571428571429
                         
), 
x = c("Amino Acids and Derivatives",
      "Cofactors, Vitamins, Prosthetic Groups",
      "Carbohydrates",
      "Fatty Acids, Lipids, and Isoprenoids",
      "Nucleosides and Nucleotides",
      "Iron acquisition and metabolism",
      "Metabolite damage and its repair or mitigation",
      "Phosphate Metabolism",
      "Nitrogen Metabolism",
      "Energy and Precursor Metabolites Generation",
      "Respiration",
      "Membrane Transport",
      "Stress Response, Defense and Virulence",
      "Protein Fate (folding, modification, targeting, degradation)",
      "Protein Synthesis",
      "Cell Cycle, Cell Division and Death",
      "Microbial communities",
      "Clustering-based subsystems",
      "Prokaryotic cell type differentiation",
      "Cell Envelope, Capsule and Slime layer",
      "RNA Processing",
      "Regulation and Cell signaling",
      "Miscellaneous",
      "Prophages, Transposable elements, Plasmids",
      "DNA Processing",
      "NA",
      "Secondary Metabolism",
      "Sulfur Metabolism"
),
Superclass=c("METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "METABOLISM",
             "ENERGY",
             "ENERGY",
             "MEMBRANE TRANSPORT",
             "STRESS RESPONSE, DEFENSE, VIRULENCE",
             "PROTEIN PROCESSING",
             "PROTEIN PROCESSING",
             "CELLULAR PROCESSES",
             "CELLULAR PROCESSES",
             "CELLULAR PROCESSES",
             "CELLULAR PROCESSES",
             "CELL ENVELOPE",
             "RNA PROCESSING",
             "REGULATION AND CELL SIGNALING",
             "MISCELLANEOUS",
             "MISCELLANEOUS",
             "DNA PROCESSING",
             "NA",
             "METABOLISM",
             "METABOLISM"
))
data1 <- data  #Comando para poner el orden de las barras, tutorial en: https://statisticsglobe.com/ordered-bars-in-ggplot2-barchart-in-r
data1$x <- factor(data1$x,
                  levels = c("Cofactors, Vitamins, Prosthetic Groups",
                             "Amino Acids and Derivatives",
                             "Carbohydrates",
                             "Fatty Acids, Lipids, and Isoprenoids",
                             "Nucleosides and Nucleotides",
                             "Secondary Metabolism",
                             "Iron acquisition and metabolism",
                             "Metabolite damage and its repair or mitigation",
                             "Sulfur Metabolism",
                             "Phosphate Metabolism",
                             "Nitrogen Metabolism",
                             "Energy and Precursor Metabolites Generation",
                             "Respiration",
                             "Membrane Transport",
                             "Stress Response, Defense and Virulence",
                             "Protein Fate (folding, modification, targeting, degradation)",
                             "Protein Synthesis",
                             "Cell Cycle, Cell Division and Death",
                             "Microbial communities",
                             "Clustering-based subsystems",
                             "Prokaryotic cell type differentiation",
                             "Cell Envelope, Capsule and Slime layer",
                             "RNA Processing",
                             "Regulation and Cell signaling",
                             "Miscellaneous",
                             "Prophages, Transposable elements, Plasmids",
                             "DNA Processing",
                             "NA"
                  ))

ggplot(data=data1,aes(y,x,fill=Superclass)) +
  geom_bar(stat="identity") +
  labs(title="Functional annotation of all isolates, average number of genes per isolate", 
       subtitle="Klebsiella pneumoniae",
       x = "Number of genes",
       y = "Class",
       caption= "") +
  scale_fill_manual(values=c("#c7f9cc",
                             "#80ed99",
                             "#57cc99",
                             "#38a3a5",
                             "#22577a",
                             "#adc900",
                             "#132a13",
                             "#0D5E30",
                             "#94d2bd",
                             "#0a9396",
                             "#e9d8a6",
                             "#90a955")) +
  theme_linedraw()

ggsave(
  filename = "Figure 1",
  plot = last_plot(),
  device = png,
  path = "/home/isaias/Desktop",
  scale = 1,
  width = 14,
  height = 7,
  units = c("in"),
  dpi = 800,
  limitsize = TRUE,
  bg = NULL
)
  
#https://ggplot2.tidyverse.org/reference/ggsave.html 
#https://towardsdatascience.com/customize-your-ggplot2-bar-graph-5-ways-to-instantly-improve-your-r-data-visualizations-f9c11dfe0163 (tutorial)
